Source/JavaScriptCore/runtime/JSGlobalObject.h

```diff
@@ -1246,6 +1250,8 @@ class JSGlobalObject : public JSSegmentedVariableObject {
 #ifdef JSC_GLIB_API_ENABLED
     std::unique_ptr<WrapperMap> m_wrapperMap;
 #endif
+
+    HashSet<RefPtr<SourceProvider>> m_sourceProviders;
```

Comment:
does this mean that for things like `eval` or any of the `WKWebView` APIs that inject some JS we would be able to see all of that even if it happens before Web Inspector was opened?  while that is a pretty neat idea, id be very concerned about webpages that aren't being debugged keeping around the source for _every_ script that's _ever_ been executed.  id definitely want some proof that this does not impact performance

Reply:
i was thinking about a `WeakHashSet`, but wouldnt the old code of walking the entire JS heap effectively do the same thing (i.e. look for anything that's still alive)?

Reply:
about `WeakHashSet`: when are sourceProviders deleted? Are they only kept around by JSFunction objects? Maybe I missed something, but I could not find a way to get sourceProviders related to other type of objects.